crates/sui-indexer/src/apis/read_api.rs

```diff
@@ -72,13 +72,15 @@ impl<S: IndexerStore> ReadApi<S> {
         let move_call_seq_number = self
             .state
             .get_move_call_sequence_by_digest(cursor.map(|digest| digest.to_string()), reverse)?;
+        let recipient_seq_number = self
```

Comment:
1.  `recipient_seq_number` seems to be unused within this function?
2. Right now we are making unnecessary db calls when these sequence numbers are only used in certain query types. For example, we should only query `move_call_seq_number` when it's a query for MoveFunction

Reply:
The second comment seems to be unresolved